app/concepts/reservations/use_cases/update.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Reservations
+  module UseCases
+    class Update
+      def initialize(id:, params:, repository: Reservations::Repository.new)
+        @id = id
+        @repository = repository
+        @params = params
+      end
+
+      def call
+        raise ActiveRecord::RecordNotFound if repository.find(id).present? == false
```

Comment:
```suggestion
        raise ActiveRecord::RecordNotFound if repository.find(id).present? == false
```
raise unless ActiveRecord::RecordNotFound if repository.find(id).present?